src/types/response.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::HashMap;
+
+use actix_http::{body::BoxBody, StatusCode};
+use actix_web::{HttpRequest, HttpResponse, HttpResponseBuilder, Responder};
+use pyo3::{
+    exceptions::{PyIOError, PyValueError},
+    prelude::*,
+    types::{PyBytes, PyDict, PyString},
+};
+
+use crate::io_helpers::{apply_hashmap_headers, read_file};
+
+use super::{check_body_type, get_body_from_pyobject};
+
+#[derive(Debug, Clone, FromPyObject)]
+pub struct Response {
+    pub status_code: u16,
+    pub response_type: String,
+    pub headers: HashMap<String, String>,
+    #[pyo3(from_py_with = "get_body_from_pyobject")]
```

Comment:
One last question - what does `from_py_with` do?

Reply:
I use this to use a method when converting the `body` field of the Python object when it is extracted to a `Response`. This way we can ensure it is either a `str` or `bytes` and convert it to a `Vec<u8>`.
You can read more here -> https://pyo3.rs/v0.18.1/conversions/traits#derivefrompyobject-field-attributes

Reply:
It is used with the `#[derive(FromPyObject)]` macro

Reply:
Perfect. Thank you! :D